spring-data-r2dbc/src/test/java/org/springframework/data/r2dbc/query/QueryMapperUnitTests.java

```diff
@@ -427,19 +440,44 @@ void mapQueryForEnumArrayShouldMapToStringList() {
 		assertThat(bindings.getCondition()).hasToString("person.enum_value IN (?[$1], ?[$2])");
 	}
 
+	@Test // gh-733
```

Comment:
These test are in the R2DBC part, but the changes are for JDBC.

Reply:
Actually, the changes are in Relational but we don't have a query mapper for Spring Data Relational and we never ensured in R2DBC how `isTrue` is bound in R2DBC. The JDBC counterpart should be in #908.

The changes in the `Criteria` API are now covered with additional assertions and the JDBC change went into effect with #908.